.github/workflows/tests.yml

```diff
@@ -20,6 +20,13 @@ jobs:
       - uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
+      - uses: astral-sh/setup-uv@v4
+        with:
+          enable-cache: true
+      - run: echo "UV_PROJECT_ENVIRONMENT=$(python -c "import sysconfig; print(sysconfig.get_config_var('prefix'))")" >> $GITHUB_ENV
+      - run: echo $UV_PROJECT_ENVIRONMENT
+      - run: uv python pin ${{ matrix.python-version }}
```

Comment:
Can you put a comment linking https://github.com/astral-sh/uv/issues/11389, something like `# uv requires .python-version to match OS Python: https://github.com/astral-sh/uv/issues/11389`